packages/block-editor/src/components/default-block-appender/index.js

```diff
@@ -75,22 +75,22 @@ export function DefaultBlockAppender( {
 export default compose(
 	withSelect( ( select, ownProps ) => {
 		const {
+			getBlockAttributes,
 			getBlockCount,
 			getBlockName,
-			isBlockValid,
 			getSettings,
 			getTemplateLock,
 		} = select( blockEditorStore );
 
 		const isEmpty = ! getBlockCount( ownProps.rootClientId );
-		const isLastBlockDefault =
+		const isLastBlockEmptyDefault =
 			getBlockName( ownProps.lastBlockClientId ) ===
-			getDefaultBlockName();
-		const isLastBlockValid = isBlockValid( ownProps.lastBlockClientId );
+				getDefaultBlockName() &&
+			getBlockAttributes( ownProps.lastBlockClientId )?.content === '';
```

Comment:
Should we use `isUnmodifiedDefaultBlock` here?

Reply:
(No guarantee that the default block has the content attribute and that it means the blocks is empty.)